Exception "java.lang.ClassCastException: android.view.ContextThemeWrapper cannot be cast to android.app.Activity ....

```
2019-11-13 19:04:40.387 17152-17152/com.ipstrust.india E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.ipstrust.india, PID: 17152
    java.lang.RuntimeException: Unable to start activity ComponentInfo{com.ipstrust.india/com.ipstrust.india.EventPostAdminActivity}: 

android.view.InflateException: Binary XML file line #11: Binary XML file line #11: Error inflating class com.chinalwb.are.AREditor

        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2914)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3049)
        at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:78)
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:108)
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:68)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1809)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loop(Looper.java:193)
        at android.app.ActivityThread.main(ActivityThread.java:6692)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
     **Caused by: android.view.InflateException: Binary XML file line #11: Binary XML file line #11: Error inflating class com.chinalwb.are.AREditor
     Caused by: android.view.InflateException: Binary XML file line #11: Error inflating class com.chinalwb.are.AREditor**
     Caused by: java.lang.reflect.InvocationTargetException
        at java.lang.reflect.Constructor.newInstance0(Native Method)
        at java.lang.reflect.Constructor.newInstance(Constructor.java:343)
        at android.view.LayoutInflater.createView(LayoutInflater.java:647)
        at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:790)
        at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:730)
        at android.view.LayoutInflater.rInflate(LayoutInflater.java:863)
        at android.view.LayoutInflater.rInflateChildren(LayoutInflater.java:824)
        at android.view.LayoutInflater.inflate(LayoutInflater.java:515)
        at android.view.LayoutInflater.inflate(LayoutInflater.java:423)
        at android.view.LayoutInflater.inflate(LayoutInflater.java:374)
        at androidx.appcompat.app.AppCompatDelegateImpl.setContentView(AppCompatDelegateImpl.java:555)
        at androidx.appcompat.app.AppCompatActivity.setContentView(AppCompatActivity.java:161)
        at com.ipstrust.india.EventPostAdminActivity.onCreate(EventPostAdminActivity.java:26)
        at android.app.Activity.performCreate(Activity.java:7140)
        at android.app.Activity.performCreate(Activity.java:7131)
        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1272)
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2894)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3049)
        at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:78)
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:108)
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:68)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1809)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loop(Looper.java:193)
        at android.app.ActivityThread.main(ActivityThread.java:6692)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)

     **Caused by: java.lang.ClassCastException: android.view.ContextThemeWrapper cannot be cast to android.app.Activity**

        at com.chinalwb.are.styles.toolbar.ARE_Toolbar.<init>(ARE_Toolbar.java:357)
        at com.chinalwb.are.styles.toolbar.ARE_Toolbar.<init>(ARE_Toolbar.java:352)
        at com.chinalwb.are.styles.toolbar.ARE_Toolbar.<init>(ARE_Toolbar.java:348)
        at com.chinalwb.are.AREditor.initGlobal(AREditor.java:156)
        at com.chinalwb.are.AREditor.init(AREditor.java:148)
2019-11-13 19:04:40.388 17152-17152/com.ipstrust.india E/AndroidRuntime:     at com.chinalwb.are.AREditor.<init>(AREditor.java:135)
        at com.chinalwb.are.AREditor.<init>(AREditor.java:122)
        	... 28 more
```


**XML**
```
<RelativeLayout xmlns:android="http://schemas.android.com/apk/res/android"
    xmlns:are="http://schemas.android.com/apk/res-auto"
    android:layout_width="match_parent"
    android:layout_height="match_parent"
    android:background="@android:color/holo_blue_dark"
    android:theme="@style/Theme.MaterialComponents"
    >



    <com.chinalwb.are.AREditor
        android:id="@+id/areditor"
        android:layout_width="match_parent"
        android:layout_height="400dp"
        android:background="@android:color/holo_green_dark"
        are:expandMode="MIN"
        are:hideToolbar="true"
        are:toolbarAlignment="TOP" />

    <com.google.android.material.button.MaterialButton
        android:layout_width="wrap_content"
        android:layout_height="wrap_content"
        android:text="POST"
        android:layout_alignParentBottom="true"
        android:layout_centerInParent="true"
        android:id="@+id/post"
        />

</RelativeLayout>



```
**JAVA CODE**



```
public class EventPostAdminActivity extends AppCompatActivity {
    AREditor arEditor;
    MaterialButton postBt;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_event_post_admin);
        arEditor = this.findViewById(R.id.areditor);
        postBt = findViewById(R.id.post);

        arEditor.setExpandMode(AREditor.ExpandMode.FULL);
        arEditor.setHideToolbar(false);
        arEditor.setToolbarAlignment(AREditor.ToolbarAlignment.BOTTOM);

        

        postBt.setOnClickListener( (view) -> System.out.println("HTML :: " + arEditor.getHtml()));

    }

   
}
```
```
```

--- a/ARE/are/src/main/java/com/chinalwb/are/Util.java
+++ b/ARE/are/src/main/java/com/chinalwb/are/Util.java
@@ -4,6 +4,7 @@ import android.annotation.SuppressLint;
 import android.app.Activity;
 import android.content.ContentUris;
 import android.content.Context;
+import android.content.ContextWrapper;
 import android.database.Cursor;
 import android.graphics.Bitmap;
 import android.graphics.Canvas;
@@ -150,6 +151,9 @@ public class Util {
     public static int getPixelByDp(Context context, int dp) {
         int pixels = dp;
         DisplayMetrics displayMetrics = new DisplayMetrics();
+        while (!(context instanceof Activity)) {
+            context = ((ContextWrapper)context).getBaseContext();
+        }
         ((Activity) context).getWindowManager().getDefaultDisplay().getMetrics(displayMetrics);
         pixels = (int) (displayMetrics.density * dp + 0.5);
         return pixels;

